File: Algorithms/438. Find All Anagrams in a String/php/solution.php
<?php
class Solution
{

    /**
     * @param String $s
     * @param String $p
     * @return Integer[]
     */
    function findAnagrams($s, $p)
    {
        $ns = strlen($s);
        $np = strlen($p);
        if ($ns < $np) {
            return [];
        }

        $pCount = array_fill(0, 26, 0);
        $sCount = array_fill(0, 26, 0);
        for ($i = 0; $i < $np; $i++) {
            $pCount[ord($p[$i]) - ord('a')]++;
        }

        $output = [];
        for ($i = 0; $i < $ns; $i++) {
            $sCount[ord($s[$i]) - ord('a')]++;
            if ($i >= $np) {
                $sCount[ord($s[$i - $np]) - ord('a')]--;
            }
            if ($sCount == $pCount) {
                $output[] = $i - $np + 1;
            }
        }

        return $output;
    }
}

File: Algorithms/438. Find All Anagrams in a String/golang/solution.go
func findAnagrams(s string, p string) []int {
	ns, np := len(s), len(p)
	if ns < np {
		return []int{}
	}

	pCount, sCount := [26]int{}, [26]int{}
	for _, ch := range p {
		pCount[ch-'a']++
	}

	output := []int{}
	for i := 0; i < ns; i++ {
		sCount[s[i]-'a']++
		if i >= np {
			sCount[s[i-np]-'a']--
		}
		if sCount == pCount {
			output = append(output, i-np+1)
		}
	}
	return output
}
